migrator/etcd.go

```diff
@@ -0,0 +1,60 @@
+package migrator
+
+import (
+	"crypto/tls"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	etcdclientv3 "go.etcd.io/etcd/clientv3"
+)
+
+const (
+	dialTimeout = time.Minute
+)
+
+func createEtcdClient(caFile string, certFile string, keyFile string, endpoint string) (*etcdclientv3.Client, error) {
+	etcdCertPair, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	etcdCaCert, err := CertPoolFromFile(caFile)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	config := etcdclientv3.Config{
+		DialTimeout: dialTimeout,
+		Endpoints: []string{
+			endpoint,
+		},
+
+		TLS: &tls.Config{
+			Certificates:       []tls.Certificate{etcdCertPair},
+			ClientCAs:          etcdCaCert,
+			RootCAs:            etcdCaCert,
+			InsecureSkipVerify: false,
+		},
+	}
+
+	client, err := etcdclientv3.New(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return client, nil
+}
+
+func etcdPeerName(index int, baseDomain string) string {
+	return fmt.Sprintf("https://etcd%d.%s:2380", index, baseDomain)
+}
+
+func initialCluster(startingIndex int, baseDomain string, nodesCount int) string {
```

Comment:
I feel like things like this deserve unit tests.

Reply:
done